pyproject.toml

```diff
@@ -114,15 +114,61 @@ test = [
 build = [
     "build>=1.2.2.post1",
     "twine>=6.1.0",
+    "versioningit>=3.3.0",
 ]
 
-[tool.setuptools.packages.find]
-where = ["."]
-include = ["disnake*"]
-
 [tool.uv]
 required-version = ">=0.9.2"
 
+[tool.hatch.build]
+include = [
+    "disnake/",
+]
+artifacts = ["disnake/_version.py"]
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "scripts/versioning.py"
+]
+
+[tool.hatch.version]
+source = "versioningit"
+
+[tool.versioningit]
+default-version = "0.0.0"
+
+[tool.versioningit.vcs]
+method = "git-archive"
+describe-subst = "$Format:%(describe:tags,match=v*)$"
+
+[tool.versioningit.format]
+distance = "{base_version}+{distance}.{vcs}{rev}"
+dirty = "{base_version}+d{build_date:%Y%m%d}"
+distance-dirty = "{base_version}.a{distance}+{vcs}{rev}.d{build_date:%Y%m%d}"
```

Comment:
These don't quite match, making builds of dirty trees use a slightly different format:
```
clean: disnake-2.11.0+66.g3fe0fa63-py3-none-any.whl
dirty: disnake-2.11.0a66+g3fe0fa63.d20251017-py3-none-any.whl
```
(note the `+66` vs `a66`)

Reply:
Which do you prefer we use?